python/ray/util/placement_group.py

```diff
@@ -123,31 +172,46 @@ def _get_none_zero_resource(self, bundle: List[Dict]):
                 return key, value
         assert False, "This code should be unreachable."
 
-    def _fill_bundle_cache_if_needed(self):
+    def _fill_bundle_cache_if_needed(self) -> None:
         if not self.bundle_cache:
-            # Since creating placement group is async, it is
-            # possible table is not ready yet. To avoid the
-            # problem, we should keep trying with timeout.
-            TIMEOUT_SECOND = 30
-            WAIT_INTERVAL = 0.05
-            timeout_cnt = 0
-            worker = ray.worker.global_worker
-            worker.check_connected()
-
-            while timeout_cnt < int(TIMEOUT_SECOND / WAIT_INTERVAL):
-                pg_info = ray.state.state.placement_group_table(self.id)
-                if pg_info:
-                    self.bundle_cache = list(pg_info["bundles"].values())
-                    return
-                time.sleep(WAIT_INTERVAL)
-                timeout_cnt += 1
-
-            raise RuntimeError(
-                "Couldn't get the bundle information of placement group id "
-                f"{self.id} in {TIMEOUT_SECOND} seconds. It is likely "
-                "because GCS server is too busy.")
+            self.bundle_cache = _get_bundle_cache(self.id)
+
 
+@client_mode_wrap
+def _call_placement_group_ready(pg_id: PlacementGroupID,
+                                timeout_seconds: int) -> bool:
+    worker = ray.worker.global_worker
+    worker.check_connected()
 
+    return worker.core_worker.wait_placement_group_ready(
+        pg_id, timeout_seconds)
+
+
+@client_mode_wrap
```

Comment:
(let me know if i'm misunderstanding what this annotation does)

why do we need to run this in a remote function? this looks like a very cheap function, so it's probably faster to run it than create a task.

Reply:
The logic here communicates with the GCS and so has to be executed server-side on the Ray cluster when using Ray client.
So `client_mode_wrap` executes it as a remote task when using the client. 

Otherwise, it just gets run on the Ray node.

Reply:
But the ray client server is a driver in the cluster so it should be able to run these things right?

Reply:
I think it's possible that this will execute on the client side if you call the public API `pg.bundle_count`.